src/check-custom-fields.ts

```diff
@@ -0,0 +1,64 @@
+import * as core from '@actions/core'
+
+interface ObjectType {
+  [key: string]: string
+}
+
+interface OptionType {
+  id: string
+  name: string
+}
+
+interface ArrayType {
+  options: OptionType[]
+  name: string
+  dataType: string
+  id: string
+}
+
+export async function checkDictionaryInArray(obj: {[key: string]: string}, arr: []) {
+  const object: ObjectType = obj
+  const array: ArrayType[] = arr
+
+  const result = []
+
+  //remove empty objects from array of objects
+  const filteredArray = array.filter(objs => !(objs && Object.keys(objs).length === 0 && objs.constructor === Object))
+
+  for (const key in object) {
+    const found = filteredArray.find(item => item.name.toLowerCase() === key.toLowerCase())
```

Comment:
Let's use some gorgeous JavaScript syntax to make this more legible

```suggestion
  for (const [fieldName, fieldValue] of Object.entries(fieldValues)) {
    const foundField = clearedProjectFields.find(projectField => projectField.name.toLowerCase() === fieldName.toLowerCase())
```